pkg/domain/entities/types/containers.go

```diff
@@ -36,6 +36,7 @@ type ContainerStatsReport struct {
 }
 
 type ContainerUpdateOptions struct {
-	NameOrID string
-	Specgen  *specgen.SpecGenerator
+	NameOrID                        string
+	Specgen                         *specgen.SpecGenerator
+	ChangedHealthCheckConfiguration *define.UpdateHealthCheckConfig
```

Comment:
I created another type where I can better distinguish what value has been set as new by the user using pointers. The entire `LinuxResources` type uses pointers or lists, but the `HealthCheck` type does not contain pointers, so I can't distinguish whether the user set the default values of the types or not.


The implementation of the `podman update`  replaces LinuxResources from the specgen without caring about the previous configuration or whether the user set the value. This will cause the resource configuration to be overwritten by bad values.

Here is an example:  
```
./bin/podman run -dt --replace --name hc1 quay.io/libpod/alpine:latest top
./bin/podman inspect hc1 --format {{.HostConfig.PidsLimit}}
# 2048
./bin/podman update hc1 --pids-limit 1024
./bin/podman inspect hc1 --format {{.HostConfig.PidsLimit}}
# 1024
./bin/podman update hc1 --cpus 2 # Change any other resource.
./bin/podman inspect hc1 --format {{.HostConfig.PidsLimit}}
# 0  <— this value should be 1024
```
I think this is a bug. I will create an issue. And I'll fix it in another PR.

Reply:
The Podman REST API has never used `spacegen` for updates.

Reply:
- https://github.com/containers/podman/issues/24610